File: Java/TextConverter/src/main/java/tddmicroexercises/textconvertor/StringEscapeUtils.java
package tddmicroexercises.textconvertor;

class StringEscapeUtils {

    public static String escapeHtml(String input) {
        String output = input;
        output = output.replace("&", "&amp;");
        output = output.replace("<", "&lt;");
        output = output.replace(">", "&gt;");
        output = output.replace("\"", "&quot;");
        output = output.replace("'", "&quot;");
        return output;
    }

}

File: kotlin/TextConverter/src/main/kotlin/tddmicroexercises/textconvertor/StringEscapeUtils.kt
package tddmicroexercises.textconvertor

internal object StringEscapeUtils {

    fun escapeHtml(input: String): String {
        var output = input
        output = output.replace("&", "&amp;")
        output = output.replace("<", "&lt;")
        output = output.replace(">", "&gt;")
        output = output.replace("\"", "&quot;")
        output = output.replace("'", "&quot;")
        return output
    }

}
